grpc/src/main/java/com/linecorp/armeria/server/grpc/GrpcServiceBuilder.java

```diff
@@ -443,6 +447,20 @@ public GrpcServiceBuilder unframedGrpcErrorHandler(UnframedGrpcErrorHandler unfr
         return this;
     }
 
+    /**
+     * Set a custom HTTP status code mapper. This is useful to serve custom HTTP status code when using unframed
+     * gRPC service.
+     * @param unframedGrpcStatusMappingFunction The function which maps the {@link Throwable} or
+     *                                          gRPC {@link Status} code to an {@link HttpStatus} code.
+     */
+    @UnstableApi
+    public GrpcServiceBuilder unframedGrpcStatusMapping(
+            UnframedGrpcStatusMappingFunction unframedGrpcStatusMappingFunction) {
+        requireNonNull(unframedGrpcStatusMappingFunction, "unframedGrpcStatusMappingFunction");
+        this.unframedGrpcStatusMappingFunction = unframedGrpcStatusMappingFunction;
+        return this;
+    }
```

Comment:
On second thought, I feel that it is not a good idea that `GrpcServiceBuilder` takes `UnframedGrpcStatusMappingFunction` as an option.
`UnframedGrpcStatusMappingFunction` is only used by `UnframedGrpcErrorHandler`.
It would be nice to provide a way to create `UnframedGrpcErrorHandler` rather than configure it with `GrpcServiceBuilder`.

```java
GrpcService.buider()
           .unframedGrpcErrorHandler(UnframedGrpcErrorHandler.of(myUnframedGrpcStatusMappingFunction))
           // Use UnframedGrpcStatusMappingFunction.of()
           .unframedGrpcErrorHandler(UnframedGrpcErrorHandler.of())
           // Use fully customized UnframedGrpcErrorHandler
           .unframedGrpcErrorHandler(myUnframedGrpcErrorHandler)
           ...
```
Consequently:
- No need to change the signature of `UnframedGrpcErrorHandler.handle()`.
- No runtime validation for mutual exclusion.
- 
@jupiny What do you think of that?

Reply:
Oh, absolutely looks better than the current implementation. 
Let me try.

Reply:
Updated!